test/es-module/test-esm-loader-mock.mjs

```diff
@@ -1,9 +1,73 @@
-// Flags: --loader ./test/fixtures/es-module-loaders/mock-loader.mjs
+// Flags: --no-warnings
 import '../common/index.mjs';
-import assert from 'assert/strict';
+import * as fixtures from '../common/fixtures.mjs';
+import assert from 'node:assert/strict';
+import { register } from 'node:module';
+import { MessageChannel } from 'node:worker_threads';
 
-// This is provided by test/fixtures/es-module-loaders/mock-loader.mjs
-import mock from 'node:mock';
+const { port1, port2 } = new MessageChannel();
+
+register(fixtures.fileURL('es-module-loaders/mock-loader.mjs'), {
+  parentURL: import.meta.url,
+  data: { port: port2 },
+  transferList: [port2],
+});
+
+/**
+ * This is the Map that saves *all* the mocked URL -> replacement Module
+ * mappings
+ * @type {Map<string, {namespace, listeners}>}
+ */
+globalThis.mockedModules = new Map();
```

Comment:
Can we make it a random string as it was in the previous test? We'd need to pass the name to the loader.

Reply:
I don't understand; it wasn't a random string. That was the builtins test.

Reply:
Oh yeah, it wasn't a random string, I must have been thinking about a different test. That being said, it wasn't a global at all, so it's a bit different. IMHO it would make sense to use `globalThis.__mockedModule_${Math.random()}` to minimize the risks of clashing with a user variable.

Reply:
How 'bout a symbol?

Reply:
We can't pass symbols cross thread, so I'm afraid it won't help us much (unless we want to use `Symbol.for`, but that's hardly better). But if I'm wrong and symbol is possible, by all means, let's use that.

Reply:
Mm, I thought of the non-serialisable issue for cross-thread, but this is on `globalThis`, which is not shared, and this whole `mockedModule` object isn't transferred.

Reply:
`generateModule` in `mock-loader` needs to reference that global, which I don't think is possible if you use a symbol. But again, if you find a way, awesome.

Reply:
OHHHHH, i see the problem now: `import.meta.mock=globalThis.mockedModules` appended to source by the load hook, which will run in main.

Reply:
I refactored to just export the variable and have the mocked modules import it. That avoids needing to use `globalThis`.